flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/CepRuntimeContextTest.java

```diff
@@ -0,0 +1,460 @@
+				return false;
+			}
+		};
+
+		RichIterativeCondition<Integer> second = new RichIterativeCondition<Integer>() {
+			private static final long serialVersionUID = 1L;
+
+			@Override
+			public boolean filter(
+				Integer value, Context<Integer> ctx) throws Exception {
+				return false;
+			}
+		};
+
+		RichIterativeCondition<Integer> third = new RichIterativeCondition<Integer>() {
+			private static final long serialVersionUID = 1L;
+
+			@Override
+			public boolean filter(
+				Integer value, Context<Integer> ctx) throws Exception {
+				return false;
+			}
+		};
+
+		RichIterativeCondition<Integer> fourth = null;
+
+		RichCompositeIterativeCondition function = new RichCompositeIterativeCondition(first, second, third, fourth) {
+			@Override
+			public boolean filter(Object value, Context ctx) throws Exception {
+				return false;
+			}
+		};
+
+		verifyRuntimeContext(function);
+		assertTrue(first.getRuntimeContext() instanceof CepRuntimeContext);
+		assertTrue(second.getRuntimeContext() instanceof CepRuntimeContext);
+		assertTrue(third.getRuntimeContext() instanceof CepRuntimeContext);
+	}
+
+	@Test
+	public void testRichAndCondition() throws Exception {
+		RichIterativeCondition<Integer> left = new RichIterativeCondition<Integer>() {
+			private static final long serialVersionUID = 1L;
+
+			@Override
+			public boolean filter(
+				Integer value, Context<Integer> ctx) throws Exception {
+				return false;
+			}
+		};
+
+		RichIterativeCondition<Integer> right = new RichIterativeCondition<Integer>() {
+			private static final long serialVersionUID = 1L;
+
+			@Override
+			public boolean filter(
+				Integer value, Context<Integer> ctx) throws Exception {
+				return false;
+			}
+		};
+
+		RichAndCondition function = new RichAndCondition<>(left, right);
+
+		verifyRuntimeContext(function);
+		assertTrue(left.getRuntimeContext() instanceof CepRuntimeContext);
+		assertTrue(right.getRuntimeContext() instanceof CepRuntimeContext);
+	}
+
+	@Test
+	public void testRichOrCondition() throws Exception {
+		RichIterativeCondition<Integer> left = new RichIterativeCondition<Integer>() {
+			private static final long serialVersionUID = 1L;
+
+			@Override
+			public boolean filter(
+				Integer value, Context<Integer> ctx) throws Exception {
+				return false;
+			}
+		};
+
+		RichIterativeCondition<Integer> right = new RichIterativeCondition<Integer>() {
+			private static final long serialVersionUID = 1L;
+
+			@Override
+			public boolean filter(
+				Integer value, Context<Integer> ctx) throws Exception {
+				return false;
+			}
+		};
+
+		RichOrCondition function = new RichOrCondition<>(left, right);
+
+		verifyRuntimeContext(function);
+		assertTrue(left.getRuntimeContext() instanceof CepRuntimeContext);
+		assertTrue(right.getRuntimeContext() instanceof CepRuntimeContext);
+	}
+
+	@Test
+	public void testRichNotCondition() throws Exception {
+		RichIterativeCondition<Integer> original = new RichIterativeCondition<Integer>() {
```

Comment:
Extract this condition, to deduplicate the code. We do not care about the implementation of `filter` here.
This line should look like: `RichIterativeCondition<Integer> original = new TestRichIterativeCondition<Integer>();`